contract/contracts/Voting/Voting.sol

```diff
@@ -112,41 +135,184 @@ contract Voting is OwnableUpgradeable, ReentrancyGuardUpgradeable, PausableUpgra
         emit ElectionVoted(msg.sender, proposalId, candidateAddress);
     }
 
+    /// @notice Concludes a proposal and processes the results
+    /// @param proposalId ID of the proposal to conclude
+    /// @dev Only the proposal creator can conclude it. Handles tie detection for elections
     function concludeProposal(uint256 proposalId) public {
         require(proposalId < proposalCount, "Proposal does not exist");
+        require(msg.sender == proposalsById[proposalId].draftedBy, "Only the founder can conclude the proposal");
 
         Types.Proposal storage proposal = proposalsById[proposalId];
-        proposal.isActive = !proposal.isActive;
 
         if(proposal.isElection){
             uint256 winnerCount = proposal.winnerCount;
             Types.Candidate[] memory candidates = proposal.candidates;
 
+            // Sort candidates by votes in descending order
             for (uint256 i = 0; i < candidates.length; i++) {
-                    for (uint256 j = i + 1; j < candidates.length; j++) {
-                        if (candidates[i].votes < candidates[j].votes) {
-                            Types.Candidate memory temp = candidates[i];
-                            candidates[i] = candidates[j];
-                            candidates[j] = temp;
+                for (uint256 j = i + 1; j < candidates.length; j++) {
```

Comment:
maybe you can use openzeppelin library `#sort` https://docs.openzeppelin.com/contracts/5.x/api/utils#Arrays to make it easy to read

Reply:
there is no function to sort in descending order. 
i would have to first use the function and then reverse it again manually